src/arrayMethodSort.js

```diff
@@ -4,8 +4,26 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => String(a) > String(b),
+  ) {
+    let flag = false;
+
+    do {
+      flag = false;
+
+      for (let i = 1; i < this.length; i++) {
+        if (compareFunction(this[i - 1], this[i]) > 0) {
```

Comment:
The comparison function should return a numerical value (negative, zero, positive) instead of a boolean. Adjust the logic to use a comparison function that returns such values, like `localeCompare` for strings or simple subtraction for numbers.